provisioning/baremetal_pod.go

```diff
@@ -782,6 +792,8 @@ func createContainerMetal3StaticIpManager(images *Images, config *metal3iov1alph
 					"FOWNER", // Needed for setting the addr_gen_mode
 				},
 			},
+			RunAsUser:  ptr.To(ironicUserID),
+			RunAsGroup: ptr.To(ironicGroupID),
```

Comment:
This container is marked as privileged, we shouldn't/cannot use specific users with it.

Reply:
Is this actually a rule? (I don't know, I'm still trying to figure this stuff out.)

I don't know that we can rely on this to solve the problem, since we'll have to upgrade from previous releases where a bunch of files were created by root, but in principle running as the ironic user seems like a good thing to do.